### PR TITLE
Initial implementation of the Jira Servicedesk integration

### LINKED ISSUE #86: Add new jira/jira-servicedesk-incidents integration
TODO: 

- [ ] Init integration from template (see: https://github.com/sensu/catalog/blob/main/templates)
  - [ ] Add `README.md`
  - [ ] Add `sensu-integration.yaml`
  - [ ] Add `CHANGELOG.md`
- [ ] Add `logo.png`
- [ ] Add `sensu-resources.yaml`
  - [ ] [jira/jira-servicedesk](https://github.com/sensu/catalog/tree/ch/flatten-the-catalog/integrations/jira/jira-servicedesk.yaml)
  - [ ] add a pipeline resource (example: [pagerduty/pagerduty-incidents](https://github.com/sensu/catalog/blob/main/integrations/pagerduty/pagerduty-incidents/sensu-resources.yaml#L1-L21))
  - [ ] move configured filters and mutators (if any) from handler(s) to the pipeline resource

### PR DESCRIPTION
**Pull request by fguimond**
Closes #86

### CODE REVIEW

File: integrations/jira/jira-servicedesk-incidents/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+<!-- Sensu Integration description; supports markdown -->
+
+The Jira Service Management integration is a Sensu Event Pipeline that generates alerts in the Jira Service Management service.
+
+<!-- Provide a high level overview of the integration contents (e.g. checks, filters, mutators, handlers, assets, etc) -->
+
+This integration provides the following resources:
+
+* `jira-servicedesk-pipeline` [pipeline]
+* `jira-servicedesk-workflow` [workflow]
```
Comment (c-kruse) [inline integrations/jira/jira-servicedesk-incidents/README.md:12]: the [workflow] link isn't defined at the bottom. I haven't been including workflow resources, though maybe I should?

File: integrations/jira/jira-servicedesk-incidents/README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+## Setup
+
+<!-- Sensu Integration setup instructions, including Sensu agent configuration and external component configuration -->
+<!-- EXAMPLE: what configuration (if any) is required in a third-party service to enable monitoring? -->
+
+1. **Get a Jira Service Management Integration Key**
+
+   While the on-premise version of Jira uses basic HTTP authentication with only
+   a simple username and password the [cloud version requires the use of an API
+   token to authenticate][10]. To create the API token perform the following steps:
```
Comment (c-kruse) [inline integrations/jira/jira-servicedesk-incidents/README.md:32]: [10] also not defined.

Probably wanted this from the handler readme:
`[10]: https://confluence.atlassian.com/cloud/deprecation-of-basic-authentication-with-passwords-for-jira-and-confluence-apis-972355348.html`

File: integrations/jira/jira-servicedesk-incidents/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+This integration provides an alert & incident management processing pipeline for use with other monitoring integrations. By default this integration will process all events passing the [built-in `is_incident` filter][is_incident] (i.e. failing events and resolution events only). Event processing via this integration may be suppressed using [Sensu Silencing][silences] (see the [built-in `not_silenced` filter][not_silenced] for more details).
+
+This integration does not produce any events that should be processed by an alert or incident management [handler].
+
+## Reference Documentation
+
+<!-- Please provide links to any relevant reference documentation to help users learn more and/or troubleshoot this integration. -->
+
+1. Pagerduty knowledge base: ["Generate an Integration Key"][pagerduty-integration-key]
```
Comment (c-kruse) [inline integrations/jira/jira-servicedesk-incidents/README.md:68]: dd

**Comment by calebhailey** [comment]
Looks like JIRA ServiceDesk is now part of "JIRA Service Management"? 

> ## Jira Service Desk vs. Jira Service Management
> 
> To be fair, it’s not really a “this vs that” scenario, because Jira Service Management includes all the functionality of Jira Service Desk with richer ITSM capabilities. Jira Service Desk is now part of Jira Service Management.
> 
> How did we get here? We created Jira Service Desk in 2013, after we noticed that nearly 40% of our customers had adapted Jira to handle service requests. Thousands of teams managed requests through service desks, reduced manual effort, and set up workflows across their organizations in departments like HR and legal.
In 2020, we introduced Jira Service Management to further unlock development, IT operations, and business teams, so they can deliver exceptional service experiences.

https://www.atlassian.com/software/jira/service-management/features/service-desk

I'm testing this now to see if it works.

**Comment by calebhailey** [comment]
It works! 

Screen Shot 2022-04-08 at 5 39 14 PM

Ran into one small issue with the handler, fixed in: https://github.com/sensu/sensu-jira-handler/pull/20

**Comment by calebhailey** [comment]
Made several updates here: 

- [x] Added a screenshot to README.md 
- [x] s/Pagerduty/JIRA/ 😅 (copy pasta)
- [x] s/Integration Key/API Token/ (more Pagerduty copy pasta)
- [x] Remove link to private GitHub repo
- [x] Remove link to Pagerduty reference documentation 
- [x] s/Jira ServiceDesk/Jira Service Management/ 
- [x] Updated prompts 
- [x] Removed namespaces from sensu-resources.yaml 
- [x] Removed unsupported Catalog variables (double-square brackets) from sensu-resources.yaml 
- [x] Added Pipeline "provider" label for Catalog auto-complete goodness
- [x] Added post_install content 
- [ ] Update prompt for API Token to be more explicit about secrets management (use the env fallback)

Still need to update this to use the fallback secrets management flow

File: integrations/jira/jira-servicedesk-incidents/README.md (hunk 1 above)
Comment (calebhailey) [reply integrations/jira/jira-servicedesk-incidents/README.md]: Correct; we don't list workflows here because they're not a first-class API resource.

**Review by calebhailey — APPROVED** [review]
LGTM!

**Pull request merged**